ml_in_a_box.sh

```diff
@@ -1,283 +1,104 @@
-# Install software for Ubuntu 20.04 ML-in-a-Box template VM on Paperspace Core
-#
-# Uses VM supplied by engineering
-# After running, VM is templatized by them
-# Script requires sudo access
-# Installs go to standard locations, not ~/src as in previous 18.04 ML-in-a-Box template from 2018
-# (except for Miniconda, required for NVidia RAPIDS)
-# Script assumes a fresh copy of the VM, with the software to be installed not already present
-# Uses a mix of install methods: Debian repo, apt-get, pip, Miniconda, depending on software
-# Some pip installs have alternative apt-get installs, but pip versions for Python/ML tend to be newer
-# See under the individual software installs below for other detail comments
-#
-# See also
-#
-# Paperspace-internal GitHub repo for this ML-in-a-Box setup: https://github.com/Paperspace/ml-in-a-box
-#
-# Last updated: Oct 04th 2021
+#!/usr/bin/env bash
 
-### TODO ###
-
-# These should be done before the VM is released
-
-# Engineering QA the GPU/CUDA/NVidia setup, especially w.r.t.
-#  - A100 GPU & CUDA so "it just works"
-#  - Versioning (most things here are fixed version, some are not; ones not have no obvious way to specify a version)
-#  - Security (we're using installed repos with arbitrary code)
-#  - Run script in one go as the superuser? (commands require sudo; VM doesn't let me be SU)
-#  - Licensing OK (looks good to me but I am not a lawyer)
-#  - Test out final VM when built
-
-# Improvements
-
-# These improve the script but are not vital
-
-# Add PyTorch 1.9/1.10 when CUDA 11.4 is supported
-# CUDA toolkit install is not verified. Later steps use it but there is a verify install using manual steps.
-# Put NumPy back to 1.21.2 after TensorFlow 2.5.0 downgrades it: pip install numpy 1.21.2 again, and then check TensorFlow still works
-# Unclear what the lib errors are when TF is asked to show the GPU devices after install
-# NVidia RAPIDS requires own virtualenv (conda, package installs, etc.), which is not ideal when nothing else installed needs it
-
-# Working directory for script
-
-mkdir ~/src
-cd ~/src
-
-
-# GPU
-# ---
-
-# - NVidia Driver 470.63.01 -
-
-# VM is set up by engineering with latest driver as of Sep 15th 2021: 470.63.01
-
-# - CUDA 11.4 -
-
-# CUDA 11.4 is present on the VM as supplied by engineering
-
-# - cuDNN 8.2.4.15-1+cuda11.4 -
-
-# Following https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html section 2.3.4.1. Ubuntu Network Installation
+sudo apt-get update
+sudo apt-get -y upgrade
+sudo apt-get -y install \
+  gcc \
+  make \
+  pkg-config \
+  apt-transport-https \
+  ca-certificates
+
+# Blacklist nouveau and rebuild kernel initramfs
+echo "blacklist nouveau
+options nouveau modeset=0" >> blacklist-nouveau.conf
+sudo mv blacklist-nouveau.conf /etc/modprobe.d/blacklist-nouveau.conf
+sudo update-initramfs -u
+sudo reboot
+
+# Install NVIDIA Linux toolkit 510.54
+wget https://us.download.nvidia.com/XFree86/Linux-x86_64/510.54/NVIDIA-Linux-x86_64-510.54.run
```

Comment:
Can I ask where you retrieved this script?
It looks it refers to more recent version of the driver.
I've been following the guidance given by 
https://developer.nvidia.com/cuda-downloads
which seems to be lock driver versions to CUDA versions (510.39.01 -> 11.6)

Reply:
Oh yeah, so this is separate from the CUDA download. The NVIDIA Unix Drivers archive page has links to those runfiles: https://www.nvidia.com/en-us/drivers/unix/

Reply:
It does bundle CUDA 11.6 by default for the current driver version at time of writing, but that's modifiable if you choose to explicitly install a different CUDA version downstream